.github/workflows/CI.yml

```diff
@@ -274,3 +274,40 @@ jobs:
         run: |
           SET CHERE_INVOKING=1
           C:\cygwin64\bin\bash -l -c "bash dev/ci.sh"
+
+  slack-notification-on-failure:
+    name: Send slack notification on CI failure
+    needs:
+      - test-unix
+      - test-cygwin
+    if: ${{ always() && github.event_name == 'push' && github.repository == 'gap-system/gap' }}
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Get branch name
+        id: get-branch
+        run: echo ::set-output name=branch::${GITHUB_REF#refs/*/}
+      - name: Get workflow status
+        id: get-status
+        shell: bash
+        run: |
+          status="success"
+          if [[ ${{ needs.test-unix.result }} = "failure" || ${{ needs.test-cygwin.result }} = "failure" ]]; then
+            status="failure"
+          fi
+          echo ::set-output name=status::$status
+      - name: Determine whether CI status changed
+        uses: gap-actions/should-i-notify-action@v1
+        id: should_notify
+        with:
+          branch: ${{ steps.vars.outputs.branch }}
+          needs_context: ${{ toJson(needs) }}
```

Comment:
The `needs` context is used to determine whether there was a failure. You are also computing that same thin in the `get-status` step above...

Can you perhaps add an output to `gap-actions/should-i-notify-action` which indicates the status? Then you could get rid of the `get-status` step, .

Reply:
I've updated the action accordingly.